packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2958,6 +2940,115 @@ Iterable<DiagnosticsNode> _describeRelevantUserCode(Element element) {
   return nodes;
 }
 
+
+/// Debugging message for DevTools deep links.
+///
+/// This property encloses it [value] as a Map<String, String> object. This map
+/// is composed of the following information:
+///
+/// `url` is the String representation of the deep link to DevTools.
+/// `screenId` is an optional id specifying the Flutter DevTools screen that this
+/// diagnostic pertains to.
+/// `objectId` is an optional id specifying the data object in Flutter DevTools
+/// that this diagnostic pertains to.
+class DevToolsDeepLinkProperty extends DiagnosticsProperty<Map<String, String>> {
```

Comment:
This should not be a property on Map<String, String>
instead make 
DevToolsDeepLink extend Diagnosticable.
Then it is clean enough to add the extra json you want and specify the tostring you needed.